src/lib/engines/test.engine.js

```diff
@@ -0,0 +1,75 @@
+const _ = require('lodash');
+
+const listeners = {};
+const subscriptions = {};
+
+/*
+ * This engine simply calls the locally saved function reference.
+ *
+ * You set what function to call with listen or subscribe.  You call functions with send or publish.
+ *
+ * Useful for testing bunnyhop APIs
+ */
+function TestEngine (pluginAPI) {
+  const log = pluginAPI.getPluginLogger();
+  const name = pluginAPI.getServiceName();
+  if (!listeners[name]) listeners[name] = {};
```

Comment:
Wont we always be using the serviceName? why hash it?

Reply:
did this because of a unit test that failed, something about how if you had 2 instances of bunnyhop at the same time they should each only get their respective message?  Otherwise if they share TestEngine they both would be reacting.